listenbrainz_spark/recommendations/candidate_sets.py

```diff
@@ -24,211 +24,217 @@
-                be generated. Dataframe columns can be depicted as:
-                    [
-                        'user_id', 'recording_id'
-                    ]
-            similar_artists_candidate_set_df (dataframe): Dataframe consisting of recording ids of
-                artists similar to top artists listened to by a user for all the users for whom
-                recommendations shall be generated. Columns can be depicted as:
-                    [
-                        'user_id', 'recording_id'
-                    ]
+            top_artists_candidate_set_df (dataframe): recording ids that belong to top artists
+                                                      corresponding to user ids.
+            top_similar_artists_candidate_set_df (dataframe): recording ids that belong to similar artists
+                                                              corresponding to user ids.
     """
     utils.save_parquet(top_artists_candidate_set_df, path.TOP_ARTIST_CANDIDATE_SET)
-    utils.save_parquet(similar_artists_candidate_set_df, path.SIMILAR_ARTIST_CANDIDATE_SET)
+    utils.save_parquet(top_similar_artists_candidate_set_df, path.SIMILAR_ARTIST_CANDIDATE_SET)
+
 
-def get_candidate_html_data(similar_artists_df, user_name):
-    """ Get artists similar to top artists listened to by the user. The function is invoked
+def get_candidate_html_data(top_similar_artists_df):
+    """ Get top and similar artists associated to users. The function is invoked
         when candidate set HTML is to be generated.
 
         Args:
-            similar_artists_df (dataframe): Dataframe of similar_artists.
-            user_name (str): User name of the user.
+            top_similar_artists_df: Dataframe containing artists similar to top artists
 
         Returns:
-            artists (dict): Dictionary can be depicted as:
+            user_data: Dictionary can be depicted as:
                 {
-                    'top_artists 1' : ['similar_artist 1', 'similar_artist 2' ... 'similar_artist x'],
-                    'top_artists 2' : ['similar_artist 1', 'similar_artist 2' ... 'similar_artist x'],
+                    'user 1' : ['top_artist 1', 'similar_artist 1', 'score'],
                     .
                     .
                     .
-                    'top_artists y' : ['similar_artist 1', 'similar_artist 2' ... 'similar_artist x'],
+                    'user n' : ['top_artist Y', 'similar_artist Z' ... 'score'],
                 }
     """
-    artists = defaultdict(list)
-    for row in similar_artists_df.collect():
-        artists[row.top_artist_name].append(row.similar_artist_name)
-    return artists
+    user_data = defaultdict(list)
+    for row in top_similar_artists_df.collect():
+        user_data[row.user_name].append((row.top_artist_name, row.similar_artist_name, row.score))
+    return user_data
+
 
-def save_candidate_html(user_data):
+def save_candidate_html(user_data, ti):
```

Comment:
ti is not a good name for this arg. time_to_generate ?

Reply:
Ti is not the time to generate, it is the initial time, we subtract it from final time and get the interval. 

Maybe time_initial?

Reply:
sure